### PR TITLE
Update the haddock documentation

### PR DESCRIPTION
**Pull request by harendra-kumar**
Some document updates. I have mentioned module names based on a proposed scheme, I will raise an issue for that.

### CODE REVIEW

**Comment by harendra-kumar** [comment]
fixes #20 .

File: lib/Unicode/UCD/Core.hs [hunk 1]
```diff
@@ -16,26 +16,38 @@ where
 import qualified Unicode.Internal.Generated.DerivedCoreProperties as P
 import qualified Unicode.Internal.Generated.PropList as P
 
--- | Selects Alphabetic Unicode characters defined in Unicode character
--- database.  This should be equivalent to "Data.Char.isAlpha".
+-- | Returns 'True' for alphabetic Unicode characters (lower-case, upper-case
+-- and title-case letters, plus letters of caseless scripts and modifiers
+-- letters).
+--
+-- prop> isAlpha == Data.Char.isAlpha
```
Comment (adithyaov) [inline lib/Unicode/UCD/Core.hs:23]: Wrap this in code blocks? `@`

File: lib/Unicode/UCD/Core.hs [hunk 2]
```diff
@@ -16,26 +16,38 @@ where
...
 {-# INLINE isAlpha #-}
 isAlpha :: Char -> Bool
 isAlpha = P.isAlphabetic
 
--- | Selects Uppercase Unicode characters defined in Unicode character database.
--- This should be equivalent to "Data.Char.isUpper".
+-- | Returns 'True' for upper-case or title-case letters.  Title case is used by
+-- a small number of letter ligatures like the single-character form of /Lj/.
+--
+-- prop> isUpper == Data.Char.isUpper
```
Comment (adithyaov) [inline lib/Unicode/UCD/Core.hs:32]: Wrap this in code blocks?

File: lib/Unicode/UCD/Core.hs [hunk 3]
```diff
@@ -16,26 +16,38 @@ where
...
+--
 {-# INLINE isUpper #-}
 isUpper :: Char -> Bool
 isUpper = P.isUppercase
 
--- | Selects Lowercase Unicode characters defined in Unicode character database.
--- This should be equivalent to "Data.Char.isLower".
+-- | Returns 'True' for lower-case letters.
+--
+-- prop> isLower == Data.Char.isLower
```
Comment (adithyaov) [inline lib/Unicode/UCD/Core.hs:40]: Wrap this in code blocks?

File: lib/Unicode/UCD/Core.hs [hunk 4]
```diff
@@ -16,26 +16,38 @@ where
...
 {-# INLINE isLower #-}
 isLower :: Char -> Bool
 isLower = P.isLowercase
 
--- | Selects White_Space Unicode characters defined in Unicode character
--- database.  This should be equivalent to "Data.Char.isSpace".
+-- | Returns 'True' for any whitespace characters, and the control
+-- characters @\\t@, @\\n@, @\\r@, @\\f@, @\\v@.
+--
+-- prop> isSpace == Data.Char.isSpace
```
Comment (adithyaov) [inline lib/Unicode/UCD/Core.hs:49]: Wrap this in code blocks?

Comment (adithyaov) [reply lib/Unicode/UCD/Core.hs:49]: Should we add test cases for these properties? Might be an overkill?

**Review by adithyaov — APPROVED** [review]

File: lib/Unicode/UCD/Core.hs (hunk 1 above)
Comment (harendra-kumar) [reply lib/Unicode/UCD/Core.hs:23]: It is recognized by haddock as it is and rendered in a box.

File: lib/Unicode/UCD/Core.hs (hunk 4 above)
Comment (harendra-kumar) [reply lib/Unicode/UCD/Core.hs:49]: We should add a cabal-docspec CI, it will check it automatically.

**Pull request merged**